numpy/core/src/multiarray/compiled_base.c

```diff
@@ -1541,9 +1542,13 @@ pack_inner(const char *inptr,
         vn_out -= (vn_out & 1);
         for (index = 0; index < vn_out; index += 2) {
             unsigned int r;
-            /* swap as packbits is "big endian", note x86 can load unaligned */
-            npy_uint64 a = npy_bswap8(*(npy_uint64*)inptr);
-            npy_uint64 b = npy_bswap8(*(npy_uint64*)(inptr + 8));
+            npy_uint64 a = *(npy_uint64*)inptr;
+            npy_uint64 b = *(npy_uint64*)(inptr + 8);
+            if (order == 'b') {
+                /* swap as packbits is "big endian", note x86 can load unaligned */
```

Comment:
Adjust the comment to make clear this is for the "big-endian" case.

Reply:
removed reference to endian, since the `'b'` makes the intention clear. Moved the rest of the comment to mark the intrinsic